pkg/clustermanager/manager.go

```diff
@@ -117,6 +117,9 @@ func (m *Manager) markUnavailable(clusterName string) {
 }
 
 func (m *Manager) start(ctx context.Context, cluster *v3.Cluster, controllers, clusterOwner bool) (*record, error) {
+	if cluster.DeletionTimestamp != nil {
```

Comment:
I don't think we should add this here. `UserContext` is called from a lot of places and so far the only one that seemed to have an issue is for the cleanup which is addressed below. I am concerned this might cause other bugs and affect other flows we aren't aware of or testing.

Reply:
I added this back, I think it's necessary to fix the issue of unnecessarily starting controllers since it was removed from `ToRESTConfig`. This should have the same behavior since before this it would hit the same check in `ToRESTConfig`. Please re-review.